File: yamhillia-client/src/models/Animal.ts
import Model from "./Model";
import AnimalSpecies from "./AnimalSpecies";
import Genders from "./Genders";

export default interface Animal extends Model {
  name: string;
  species: AnimalSpecies;
  gender: Genders;
  image: string | null;
  customIdentifier: string | null;
  dateOfBirth: string | null;
  dateOfDeath: string | null;
}

File: YamhilliaNET/YamhilliaNET/Models/Animal.cs
using System;
using System.ComponentModel.DataAnnotations;
using YamhilliaNET.Constants;

namespace YamhilliaNET.Models
{
    public class Animal : AbstractYamhilliaModel
    {
        [Required]
        public string Name { set; get; }

        [Required]
        public Genders Gender { set; get; }

        [Required]
        public Species Species { set; get; }

        public DateTime? DateOfBirth { set; get;}

        public DateTime? DateOfDeath { set; get;}
 
        public virtual Farm Farm { set; get; }

        public long FarmId { set; get;}

        public string CustomIdentifier { set; get;}
    }
}